nilearn/maskers/tests/test_surface_maps_masker.py

```diff
@@ -294,3 +296,206 @@ def test_surface_maps_masker_sample_mask_to_fit_transform(
     )
     # we remove two samples via sample_mask so we should have 3 samples
     assert signals.shape == (3, masker.n_elements_)
+
+
+# ------------------ Tests for reporting ------------------
+
+
+@pytest.mark.skipif(
+    not is_matplotlib_installed(),
+    reason="Test requires matplotlib to be installed.",
+)
```

Comment:
I think it may be useful to maybe use the `pyplot` fixure

https://github.com/nilearn/nilearn/blob/bc67775f6153c533e18edaabdb551b7f4c1f5e4e/nilearn/plotting/tests/conftest.py#L7

Like used here for example: 

https://github.com/nilearn/nilearn/blob/bc67775f6153c533e18edaabdb551b7f4c1f5e4e/nilearn/plotting/tests/test_surf_plotting.py#L275

it will just skip the test if matplotlib is not there

feel free to move the content from `nilearn/plotting/tests/conftest.py` to `nilearn/conftest.py`

Reply:
Nice! Is it ok if I rename `pyplot` fixture to `matplotlib`?
I got confused for a second between `plotly` and `pyplot`.

Reply:
> Is it ok if I rename `pyplot` fixture to `matplotlib`?

yup fine: though makes me realize that having a fixture with the same name as a package or a module sounds like recipe for disaster if we end up doing `import matplolib` and no alias it via `import matplotlib as mpl`

Reply:
Yeah fair point. How about matplotlib_pyplot?

Reply:
Also @Remi-Gau how did you get around ARG001 Ruff failures in [`nilearn/nilearn/plotting/tests/test_surf_plotting.py`](https://github.com/nilearn/nilearn/blob/bc67775f6153c533e18edaabdb551b7f4c1f5e4e/nilearn/plotting/tests/test_surf_plotting.py#L275)

![Screenshot 2025-01-20 at 11 10 21 AM](https://github.com/user-attachments/assets/81f3b80d-2f11-4906-a13f-822042d6c665)

Reply:
used the very unsubtle nuclear option:

https://github.com/nilearn/nilearn/blob/b851ecc5a14a6d11aa9f8c2aa59cbede493e699f/nilearn/plotting/tests/test_surf_plotting.py#L3

Reply:
Done: https://github.com/nilearn/nilearn/pull/4968/commits/fe94ca58b1223effbf54a4c625eff3b5300fa760